CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+# Contributing
+
+Contributions from the community are essential in keeping Hibernate (and any Open Source
+project really) strong and successful.  
+
+# Legal
+
+You should check the relevant license under which all contributions will be licensed for the specific project you are contributing to.
+
+All contributions are subject to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).  
+
+The DCO text is available verbatim in the [dco.txt](dco.txt) file.
+
+
+## Guidelines
+
+While we try to keep requirements for contributing to a minimum, there are a few guidelines 
+we ask that you mind.
+
+For code contributions, these guidelines include (**where applicable**):
+* Respect the project code style.
+* Have a corresponding JIRA [issue](https://hibernate.atlassian.net/browse/HHH) and be sure to include the key for this JIRA issue in your commit messages.
+* Have a set of appropriate tests.
+  	
+	When submitting bug reports, the tests should reproduce the initially reported bug and illustrate that your solution addresses the issue.
+	For features/enhancements, the tests should demonstrate that the feature works as intended.  
+    	In both cases, be sure to incorporate your tests into the project to protect against possible regressions.
+* If applicable, documentation should be updated to reflect the introduced changes
+* Make sure the code compiles and the tests pass
+
+For documentation contributions, mainly to respect the project code style, especially in regard 
+to the use of tabs - use the code style templates where available.  Ideally, these contributions would also have a corresponding JIRA issue, although this is less necessary for documentation contributions.
+
+
+## Getting Started
+
+If you are just getting started with Git, GitHub, and/or contributing to Hibernate via
+GitHub there are a few pre-requisite steps to follow:
+
+* Make sure you have a [Hibernate JIRA account](https://hibernate.atlassian.net)
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* [Fork](https://help.github.com/articles/fork-a-repo) the relevant Hibernate repository.  As discussed in
+the linked page, this also includes:
+    * [set up your local git install](https://help.github.com/articles/set-up-git) 
+    * clone your fork
+* Instruct git to ignore certain commits when using `git blame`. From the directory of your local clone, run this: `git config blame.ignoreRevsFile .git-blame-ignore-revs`
+* See the wiki pages for setting up your IDE, whether you use 
+[IntelliJ IDEA](https://hibernate.org/community/contribute/intellij-idea/)
+or [Eclipse](https://hibernate.org/community/contribute/eclipse-ide/)<sup>(1)</sup>.
```

Comment:
these IDE pages are a bit out of date; here: https://github.com/hibernate/hibernate.org/pull/230#issuecomment-2636369902 we were thinking that maybe they should go away.... so maybe let's not mention them in the new document?